libdnf/utils/fs/file.cpp

```diff
@@ -0,0 +1,294 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#include "file.hpp"
+
+#include "libdnf/common/exception.hpp"
+
+extern "C" {
+#include <solv/solv_xfopen.h>
+}
+
+#include <cstdio>
+
+
+#define libdnf_assert_file_open() libdnf_assert(file != nullptr, "The operation requires an open file");
+
+
+namespace libdnf::utils::fs {
+
+File::File(const std::filesystem::path & path, const char * mode, bool use_solv_xfopen) {
+    open(path, mode, use_solv_xfopen);
+}
+
+
+File::File(int fd, const std::filesystem::path & path, const char * mode, bool use_solv_xfopen_fd) {
+    open(fd, path, mode, use_solv_xfopen_fd);
+}
+
+
+File::File(File && f) : file(f.file) {
```

Comment:
What about the `path`?
`File::File(File && f) : path(std::move(f.path)), file(f.file) {`